File: game_server/ClientProtocol/NPEnum/ENPPlayerComboTitleType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace NPEnum
{

/// <summary>
/// 玩家组合具体称号
/// </summary>
public enum ENPPlayerComboTitleType {
	[InspectorName("PRE - idx[0] - 前缀")]
	PRE, //0 ==== 前缀
	[InspectorName("SFX - idx[1] - 后缀")]
	SFX, //1 ==== 后缀
	[InspectorName("BG - idx[2] - 底色")]
	BG, //2 ==== 底色
}

public class ENPPlayerComboTitleTypeComparer : IEqualityComparer<ENPPlayerComboTitleType>{
	public bool Equals(ENPPlayerComboTitleType x, ENPPlayerComboTitleType y) { return x == y; }
	public int GetHashCode(ENPPlayerComboTitleType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 3;
}
}

File: game_server/ServerProtocol/src/NPEnum/ENPPlayerComboTitleType.java
package NPEnum;

/*********
 * 玩家组合具体称号
 **/
public enum ENPPlayerComboTitleType {
	PRE, //0 ==== 前缀
	SFX, //1 ==== 后缀
	BG, //2 ==== 底色
	;
public static final ENPPlayerComboTitleType[]  ENPPlayerComboTitleType_Values = ENPPlayerComboTitleType.values();
public static final int ENPPlayerComboTitleType_Length = ENPPlayerComboTitleType_Values.length;
public static ENPPlayerComboTitleType ENPPlayerComboTitleType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= ENPPlayerComboTitleType_Length){ return null; }
	return ENPPlayerComboTitleType_Values[_ivalue];
}
}

